neuralforecast/common/_base_model.py

```diff
@@ -190,7 +191,11 @@ def __repr__(self):
         return type(self).__name__ if self.alias is None else self.alias
 
     def _check_exog(self, dataset):
-        temporal_cols = set(dataset.temporal_cols.tolist())
+        if isinstance(dataset, IterativeTimeSeriesDataset):
+            temporal_cols = set(dataset.files_ds.temporal_cols.tolist())
+        else:
+            temporal_cols = set(dataset.temporal_cols.tolist())
+
```

Comment:
could temporal_cols instead be set in IterativeTimeSeriesDataset as a property? see comment below re common base class